File: 27-RemoveElement/27-RemoveElement.cpp
// Last updated: 12/6/2025, 5:54:56 am
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int pos = nums.size()-1;
        for (int i=0; i<nums.size(); i++){
            if (nums[i] == val){
                nums[i] = nums[pos];
                pos--; i--;
            }if (i >= pos) break;
        }return pos+1;
    }
};

File: 27-RemoveElement/27-RemoveElement.py
# Last updated: 12/6/2025, 5:54:50 am
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        i = 0
        n = len(nums)
        j = n-1

        while i <= j:
            if nums[i] == val:
                nums[i] = nums[j]
                j -= 1
                n -= 1
            else:
                i += 1
        return n